src/circuit.ts

```diff
@@ -55,6 +56,33 @@ export class Circuit {
         return getTotalConstraints(this._circuitConfig.r1csPath)
     }
 
+    async downloadFileOverHttp (fileUrl, outputPath) {
+        
+        const file = fs.createWriteStream(outputPath);
+        return new Promise((resolve, reject) => {
+            https.get(fileUrl, function(response) {
+                response.pipe(file);
+
+                file.on("finish", () => {
+                    file.close();
+                    resolve(true)
+                });
+            });
+        })
+    }
+ 
+    async downloadPowerOfTauFile(){
+        // TODO: this url should be generated according to the total constraints
+        const tauFileUrl = "https://hermez.s3-eu-west-1.amazonaws.com/powersOfTau28_hez_final_14.ptau"
```

Comment:
create a map of constraintsToURL using the following table 
https://imgur.com/mMRG7ne

https://github.com/iden3/snarkjs#readme

Invoke circuit.ts getTotalConstraints

Also compile cannot use downloadPowerOtTau because compile is used in getTotalConstraints. This will be a recursive call

we can download these files during validation step only.

Reply:
@ivorynoise we should not download on validation, because this will download the file every time CircomJS is imported.

We have to check if a given tau file exists { we should hence have a common powers-of-tau folder inside }, if It doesn't only then we have to download it, therefore we should put it in compile only but only download if it is not present in the common powers of tau folder.

Reply:
@bajpai244 I've updated the PR.

1. Now it will check if "tau" folder is present or not in output directory. if not, it will create it.
2. Then it will check if that tau file is present or not. If not, it will download it.

Reply:
> 

We also had similar thoughts :)